File: app/schemas/conversation.py
"""Conversation request and response schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ConversationCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    title: str | None = Field(default=None, max_length=200)


class ConversationOut(BaseModel):
    conversation_id: str
    title: str | None
    created_at: datetime | None = None
    updated_at: datetime | None = None


class MessageOut(BaseModel):
    message_id: str
    role: str
    content: str
    created_at: datetime | None = None


class ConversationDetailOut(ConversationOut):
    messages: list[MessageOut] = []

File: lifeagent-frontend/src/types/conversation.ts
// 与后端 app/schemas/conversation.py 对齐（Backend API contract is the source of truth）。

export type MessageRole = 'user' | 'assistant'

export interface ConversationOut {
  conversation_id: string
  title: string | null
  created_at: string | null
  updated_at: string | null
}

export interface MessageOut {
  message_id: string
  role: MessageRole
  content: string
  created_at: string | null
}

export interface ConversationDetailOut extends ConversationOut {
  messages: MessageOut[]
}

export interface ConversationCreatePayload {
  title?: string | null
}
